client/src/main/java/ch/bifrost/client/impl/session/KeyExchangeClient.java

```diff
@@ -0,0 +1,36 @@
+package ch.bifrost.client.impl.session;
+
+import java.net.InetAddress;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Optional;
+
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.Packet;
+import ch.bifrost.core.api.session.IdKeyPair;
+import ch.bifrost.core.api.session.KeyExchange;
+
+public class KeyExchangeClient implements KeyExchange {
+
+	private DatagramEndpoint datagramEndpoint;
+	private InetAddress serverHost;
+	private int serverPort;
+
+	public KeyExchangeClient(DatagramEndpoint datagramEndpoint, InetAddress serverHost, int serverPort) {
```

Comment:
Issue #14 addresses this: the key exchange itself can be oblivious of the fact that the server is addressed by "InetAddress:Port". This can be achieved by separating the "sending endpoint" (initialized with the server's address) and the "receiving endpoint". The idea: keep the key exchange less dependent on the underlying architecture.

I don't mean to start a discussion here, just wanted to mark the position. Let's discuss in #14 or in person.